overcloud.yml

```diff
@@ -281,3 +281,12 @@
             infrared tripleo-overcloud -vvv --version {{ osp_release }} --deployment-timeout {{ deployment_timeout | default(240) }} --build {{ osp_puddle }} --deployment-files {{ nic_configs }} --introspect no --tagging no --deploy yes --controller-nodes {{ controller_count }} --compute-nodes {{ compute_count }} --storage-nodes {{ ceph_node_count }} --overcloud-image-name full {{ oc_extra_templates }} --network-protocol ipv4 --network-backend {{ network_backend }} {{ network_type }} true {{ ceph_params | default('') }} {{ storage_protocol_backend | default('') }} --ntp-server {{ gateway }} --public-network false {{ oc_heat_configs | default('') }} {{ oc_config_resource | default('') }} > {{ log_directory }}/overcloud_deploy.log 2>&1
         args:
             chdir: "{{ infrared_dir }}"
+        when: (osp_release|int < 17) or (osp_release|int == 17 and not ceph_enabled)
+
+      - name: run tripleo-overcloud deploy with ceph for osp 17
+        shell: |
+            source .venv/bin/activate
+            infrared tripleo-overcloud -vvv --version {{ osp_release }} --deployment-timeout {{ deployment_timeout | default(240) }} --build {{ osp_puddle }} --deployment-files {{ nic_configs }} --introspect no --tagging no --deploy yes --controller-nodes {{ controller_count }} --compute-nodes {{ compute_count }} --storage-nodes {{ ceph_node_count }} {{ oc_extra_templates }} --network-protocol ipv4 --network-backend {{ network_backend }} {{ network_type }} true {{ ceph_params | default('') }} {{ storage_protocol_backend | default('') }} --ceph-osd-spec-file {{ osd_spec_file }} --ntp-server {{ gateway }} --public-network false {{ oc_heat_configs | default('') }} {{ oc_config_resource | default('') }} > {{ log_directory }}/overcloud_deploy.log 2>&1
+        args:
+            chdir: "{{ infrared_dir }}"
+        when: osp_release|int == 17 and ceph_enabled
```

Comment:
Should this be changed to `osp_release|int >= 17` for now? If there are deployment changes in future releases, we can change it again later. I think it might be better to assume that there will not be deployment changes in this aspect in future releases, and in that case, we wouldn't have to change the condition check again if it is `osp_release|int >= 17`. Thoughts?

Reply:
As discussed, we should be changing all the conditions where we have used == 17 to >= 17, so i'll make this change.